core/src/main/java/tc/oc/pgm/modules/StatsMatchModule.java

```diff
@@ -167,6 +173,19 @@ public void onMatchEnd(MatchFinishEvent event) {
     }
   }
 
+  @EventHandler
+  public void onPlayerLeave(PlayerQuitEvent event) {
+    Player player = event.getPlayer();
+    if (allPlayerStats.containsKey(player.getUniqueId()))
+      cachedPlayers.put(player.getUniqueId(), player);
+  }
+
+  @EventHandler
+  public void onPlayerJoin(PlayerJoinEvent event) {
+    UUID playerUUID = event.getPlayer().getUniqueId();
+    cachedPlayers.remove(playerUUID);
```

Comment:
Do you have these two methods mixed up? You're adding to cached on leave and removing on join?

Reply:
I may have used "cahce" wrong. I need to store every player that has stats for each match that logs off in case they get a top stat, because Bukkit might lose track of them if too many new players join.
If they log back on i dont need to store them anymore.
Do you have any suggestions for something else i could call this?

Reply:
We can't keep `Player` objects in-memory after someone leaves, this will cause memory leaks. Instead, you would need to store their username, as a string.

Reply:
Ok!

Reply:
But is there another more suitable name for the storage?

Reply:
@KingOfSquares could do 
`private final Map<UUID, String> cachedUsernames = new HashMap<>();`

Reply:
Ok! I was just wondering if not to use "cache", thanks :)